src/Functions/fromReadableSize.cpp

```diff
@@ -0,0 +1,265 @@
+
+#include <Columns/ColumnsNumber.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/IFunction.h>
+#include "base/types.h"
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int TOO_FEW_ARGUMENTS_FOR_FUNCTION;
+    extern const int TOO_MANY_ARGUMENTS_FOR_FUNCTION;
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int BAD_ARGUMENTS;
+}
+
+namespace
+{
+
+    const std::unordered_map<std::string_view, UInt64> size_unit_to_bytes =
+    {
+        {"b", 1},
+        // ISO/IEC 80000-13 binary units
+        {"kib", 1024},                  // 1024
+        {"mib", 1048576},               // 1024 * 1024
+        {"gib", 1073741824},            // 1024 * 1024 * 1024
+        {"tib", 1099511627776},         // 1024 * 1024 * 1024 * 1024
+        {"pib", 1125899906842624},      // 1024 * 1024 * 1024 * 1024 * 1024
+        {"eib", 1152921504606846976},   // 1024 * 1024 * 1024 * 1024 * 1024 * 1024
+
+        // SI units
+        {"kb", 1000},                    // 10e3
+        {"mb", 1000000},                 // 10e6
+        {"gb", 1000000000},              // 10e9
+        {"tb", 1000000000000},          // 10e12
+        {"pb", 1000000000000000},       // 10e15
+        {"eb", 1000000000000000000},    // 10e18
+    };
+    constexpr UInt64 MAX_UINT64 = std::numeric_limits<UInt64>::max();
+
+    class FunctionFromReadableSize : public IFunction
+    {
+    public:
+        static constexpr auto name = "fromReadableSize";
+        static FunctionPtr create(ContextPtr) { return std::make_shared<FunctionFromReadableSize>(); }
+
+        String getName() const override { return name; }
+
+        bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return true; }
+
+        size_t getNumberOfArguments() const override { return 1; }
+
+        DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+        {
+            if (arguments.empty())
+                throw Exception(
+                    ErrorCodes::TOO_FEW_ARGUMENTS_FOR_FUNCTION,
+                    "Number of arguments for function {} doesn't match: passed {}, should be 1.",
+                    getName(),
+                    arguments.size());
+
+            if (arguments.size() > 1)
+                throw Exception(
+                    ErrorCodes::TOO_MANY_ARGUMENTS_FOR_FUNCTION,
+                    "Number of arguments for function {} doesn't match: passed {}, should be 1.",
+                    getName(),
+                    arguments.size());
+
+            const IDataType & type = *arguments[0];
+
+            if (!isString(type))
+                throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Cannot format {} as time string.", type.getName());
+
+            return std::make_shared<DataTypeUInt64>();
+        }
+
+        bool useDefaultImplementationForConstants() const override { return true; }
```

Comment:
Could we move this function up, somewhere to l. 56 for example?

Reply:
Done!